front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -507,7 +509,7 @@ export async function* runRetrieval(
         id: 0, // dummy pending database insertion
         dataSourceId: d.data_source_id,
         documentId: d.document_id,
-        reference: "",
+        reference: new_id().slice(0, 3),
```

Comment:
NIT: In the type this field is described as is: "Short random string so that the model can refer to the document."
As "reference" is not self-explanatory, what do you think about calling `sId` and have it sliced to 10 char?